airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -313,91 +413,127 @@ def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path, validator
     storage_client = storage.Client(credentials=credentials)
     bucket = storage_client.bucket(bucket_name)
     docs_path = Path(validator_opts.docs_path)
-
-    icon_uploaded, icon_blob_id = _icon_upload(metadata, bucket, icon_file_path)
+    gcp_connector_dir = f"{METADATA_FOLDER}/{metadata.data.dockerRepository}"
 
     # Upload version metadata and doc
     # If the connector is a pre-release, we use the pre-release tag as the version
     # Otherwise, we use the dockerImageTag from the metadata
-    version = metadata.data.dockerImageTag if not is_pre_release else validator_opts.prerelease_tag
-    version_uploaded, version_blob_id = _metadata_upload(metadata, bucket, metadata_file_path, version)
-    doc_version_uploaded, doc_version_blob_id = _doc_upload(metadata, bucket, docs_path, False, False)
-    doc_inapp_version_uploaded, doc_inapp_version_blob_id = _doc_upload(metadata, bucket, docs_path, False, True)
-
-    latest_uploaded, latest_blob_id = False, None
-    doc_latest_uploaded, doc_latest_blob_id = doc_inapp_latest_uploaded, doc_inapp_latest_blob_id = False, None
-
-    release_candidate_uploaded, release_candidate_blob_id = False, None
-
-    # Latest upload
-    # We upload
-    # - the current metadata to the "latest" path
-    # - the current doc to the "latest" path
-    # - the current inapp doc to the "latest" path
-    if should_upload_latest:
-        latest_uploaded, latest_blob_id = _metadata_upload(metadata, bucket, metadata_file_path, LATEST_GCS_FOLDER_NAME)
-        doc_latest_uploaded, doc_latest_blob_id = _doc_upload(metadata, bucket, docs_path, True, False)
-        doc_inapp_latest_uploaded, doc_inapp_latest_blob_id = _doc_upload(metadata, bucket, docs_path, True, True)
+    upload_as_version = metadata.data.dockerImageTag if not is_pre_release else validator_opts.prerelease_tag
+
+    # Start uploading files
+    uploaded_files = []
+
+    # Metadata upload
+    metadata_files_uploaded = _file_upload(
+        file_key="metadata",
+        local_path=metadata_file_path,
+        gcp_connector_dir=gcp_connector_dir,
+        bucket=bucket,
+        upload_as_version=upload_as_version,
+        upload_as_latest=should_upload_latest,
+        disable_cache=True,
+        override_destination_file_name=METADATA_FILE_NAME,
+    )
+    uploaded_files.extend(metadata_files_uploaded)
 
     # Release candidate upload
     # We just upload the current metadata to the "release_candidate" path
     # The doc and inapp doc are not uploaded, which means that the release candidate will still point to the latest doc
```

Comment:
❗ I think this comment means we need to add some logic around release candidacy (and probably pre-release?) to not upload docs for them, if we want parity with the old behavior. presumably we also probably dont want the icon uploaded unless we're uploading to latest, since it's not versioned. 

Though I'm confused why this was intentional in the first place, if it's going to its own filepath, i feel like it should be okay to get its own docs that are loaded up if a user is using that release candidate version

Reply:
Or maybe since these files are going somwhere else it doesn't matter? I am a little lost on this bc im out of the loop with release candidate stuff, so i'll let @alafanechere chime in here

Reply:
The tests should cover this more explicitly now